CodeConverter/CSharp/HoistedFieldFromVbStaticVariable.cs

```diff
@@ -6,18 +6,21 @@ internal class HoistedFieldFromVbStaticVariable : IHoistedNode
 {
     public string OriginalMethodName { get; }
     public string OriginalVariableName { get; }
+    public string OriginalParentAccessorKind { get; }
     public ExpressionSyntax Initializer { get; }
     public TypeSyntax Type { get; }
     public bool IsStatic { get; }
 
-    public HoistedFieldFromVbStaticVariable(string originalMethodName, string originalVariableName, ExpressionSyntax initializer, TypeSyntax type, bool isStatic)
+    public HoistedFieldFromVbStaticVariable(string originalMethodName, string originalVariableName, string originalParentAccessorKind, ExpressionSyntax initializer, TypeSyntax type, bool isStatic)
     {
         OriginalMethodName = originalMethodName;
         OriginalVariableName = originalVariableName;
+        OriginalParentAccessorKind = originalParentAccessorKind;
```

Comment:
As a general rule, try to keep strong typing where it exists. Much easier for the receiver to know they have that enum and can do whatever with it rather than just an opaque string.

In the general case, I mean keep an enum or a small wrapper, I don't mean keep the whole syntax node around.

If keeping it as a string here feels right, then the name is probably wrong, and it should just be something like UniquePrefixFromContext here. Either way is OK, it's just this walks the middle path where both the caller and consumer are making half the decision about how to use this.

Reply:
I chose string because `SyntaxAnnotation.Data` is a string. And I thought I'd avoid parsing the data back to an enum, but it doesn't happen anyway with the way the code currently is, so I replaced it with the `MethodKind` enum now.

Reply:
The choice is between `VBasic.SyntaxKind`, `MethodKind` and some new enum just for that field... If you prefer another option, I'll change it.

Reply:
Looks good, thanks!